catroid/src/main/java/org/catrobat/catroid/ui/adapter/BrickAdapter.java

```diff
@@ -1064,7 +1052,34 @@ public void onClick(DialogInterface dialog, int item) {
 		}
 	}
 
-	protected void copyBrickListAndProject(int itemPosition) {
+	private void launchAddBrickAndSelectBrickAt(Context context, int index) {
+		if (!viewSwitchLock.tryLock()) {
```

Comment:
There's actually a lock with a timeout of 200 milliseconds before it is unlocked again --> have a look at our ViewSwitchLock implementation.